contracts/ERC721Collateralizer.sol

```diff
@@ -0,0 +1,372 @@
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity 0.4.18;
+
+import "zeppelin-solidity/contracts/math/SafeMath.sol";
+import "zeppelin-solidity/contracts/token/ERC721/ERC721.sol";
+
+import "./TermsContract.sol";
+import "./DebtRegistry.sol";
+import "./ERC721TokenRegistry.sol";
+import {PermissionsLib, PermissionEvents} from "./libraries/PermissionsLib.sol";
+
+
+/**
+  * Contains functionality for collateralizing NFTs, by transferring them from
+  * a debtor address to this contract as a custodian.
+  */
+contract ERC721Collateralizer is Pausable, PermissionEvents {
+    using PermissionsLib for PermissionsLib.Permissions;
+    using SafeMath for uint;
+
+    address public debtKernelAddress;
+
+    DebtRegistry public debtRegistry;
+    ERC721TokenRegistry public tokenRegistry;
+
+    // Collateralizer here refers to the owner of the asset that is being collateralized.
+    mapping(bytes32 => address) public agreementToCollateralizer;
+
+    PermissionsLib.Permissions internal collateralizationPermissions;
+
+    uint public constant SECONDS_IN_DAY = 24 * 60 * 60;
+
+    string public constant CONTEXT = "ERC721Collateralizer";
+
+    event CollateralLocked(
+        bytes32 indexed agreementID,
+        // The address of the ERC721 contract for the token.
+        address indexed token,
+        // The ID of the token in the ERC721 contract.
+        uint256 tokenID
+    );
+
+    event CollateralReturned(
+        bytes32 indexed agreementID,
+        address indexed collateralizer,
+        address token,
+        uint256 tokenID
+    );
+
+    event CollateralSeized(
+        bytes32 indexed agreementID,
+        address indexed beneficiary,
+        address token,
+        uint256 tokenID
+    );
+
+    modifier onlyAuthorizedToCollateralize() {
+        require(collateralizationPermissions.isAuthorized(msg.sender));
+        _;
+    }
+
+    function ERC721Collateralizer(
+        address _debtKernel,
+        address _debtRegistry,
+        address _tokenRegistry
+    ) public {
+        debtKernelAddress = _debtKernel;
+        debtRegistry = DebtRegistry(_debtRegistry);
+        tokenRegistry = ERC721TokenRegistry(_tokenRegistry);
+    }
+
+    /**
+     * Transfers collateral from the debtor to the current contract, as custodian.
+     *
+     * @param agreementId bytes32 The debt agreement's ID
+     * @param collateralizer address The owner of the asset being collateralized
+     */
+    function collateralize(
+        bytes32 agreementId,
+        address collateralizer
```

Comment:
this variable name confuses me a bit -- because i think of this contract as the collateralizer, which also has an address